Remove jakarta.json.Json usage for performance reasons

peuBouzon opened the pull request:
related to #42748

peuBouzon committed:
remove jakarta.json.Json usage for performance reasons

jmartisk commented on extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java:
```diff
@@ -8,19 +10,14 @@
-import graphql.ExecutionResultImpl;
-import graphql.GraphQLError;
-import graphql.GraphqlErrorBuilder;
+import graphql.*;
```
Please avoid star imports

jmartisk commented on extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/JsonProviderProvider.java:
```diff
@@ -0,0 +1,7 @@
+
+import jakarta.json.spi.JsonProvider;
+
+public class JsonProviderProvider {
```
Would `JsonProviderHolder` sound less silly? :)

gsmet replied:
+1 for the name change but I also think this class should be moved to the `quarkus-jsonp` extension so that we can share this globally.

It should be put in the runtime module in the `io.quarkus.jsonp` package (as it's more of an API).

gsmet reviewed (changes requested):
The principle looks good. Thanks for this work.

I added a suggestion though.

peuBouzon committed:
remove star imports

peuBouzon committed:
move JsonProviderHolder to quarkus-jsonp

gsmet commented on extensions/jsonp/runtime/src/main/java/io/quarkus/jsonp/runtime/JsonProviderHolder.java:
```diff
@@ -0,0 +1,11 @@
+package io.quarkus.jsonp.runtime;
```
I specifically asked for the `io.quarkus.jsonp` package in my comment because I think we will consider this API.

Could you adjust, please?

There is a somewhat unwritten rule that  `io.quarkus.<extension>.runtime` is for internal classes.

jmartisk replied:
So this will be considered public API?
In that case it may warrant at least a mention somewhere in the docs. Or at minimum a javadoc comment.

peuBouzon committed:
move JsonProviderHolder from io.quarkus.jsonp.runtime to io.quarkus.jsonp

peuBouzon committed:
add a javadoc to JsonProviderHolder

gsmet committed:
Adjust the API a bit

And remove one more JsonProvider initialization in Micrometer.

gsmet committed:
Add a note in the doc for JsonProvider

Note that it's hard to find a place to put it given most of the time you
shouldn't use this API directly.

gsmet commented:
@peuBouzon @jmartisk I pushed two additional commits:

- the first one make sure we don't use a constant. Using constants in API is problematic as you won't be able to change the implementation if we need to
- the second one adds some warnings in the doc

If all is good for you, I think we can merge once CI is green.

gsmet commented:
Thanks @peuBouzon , nice team work!
